tests/foreman/maintain/test_advanced.py

```diff
@@ -0,0 +1,434 @@
+    """Verify that all required repositories gets enabled.
+
+    :id: e32fee2d-2a1f-40ed-9f94-515f75511c5a
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run repositories-setup --version 6.y
+
+    :BZ: 1684730, 1869731
+
+    :expectedresults: Required Satellite repositories for install/upgrade should get enabled
+    """
+    supported_versions = ['6.11', '6.12']
+    for ver in supported_versions:
+        result = sat_maintain.cli.Advanced.run_repositories_setup(options={'version': ver})
+        assert 'FAIL' not in result.stdout
+        assert result.status == 0
+        result = sat_maintain.execute('yum repolist')
+        for repo in sat_repos[ver]:
+            assert repo in result.stdout
+
+    # 6.13 till not GA
+    result = sat_maintain.cli.Advanced.run_repositories_setup(options={'version': '6.13'})
+    assert 'FAIL' in result.stdout
+    assert result.status == 1
+    for repo in sat_repos['6.13']:
+        assert repo in result.stdout
+
+    # Verify that all required beta repositories gets enabled
+    # maintain beta repo is unavailable for EL8 https://bugzilla.redhat.com/show_bug.cgi?id=2106750
+    sat_beta_repo = ['satellite-6-beta-for-rhel-8-x86_64-rpms'] + common_repos
+    missing_beta_el8_repos = ['satellite-maintenance-6-beta-for-rhel-8-x86_64-rpms']
+    result = sat_maintain.cli.Advanced.run_repositories_setup(
+        options={'version': '6.12'}, env_var='FOREMAN_MAINTAIN_USE_BETA=1'
+    )
+    assert 'FAIL' in result.stdout
+    assert result.status != 0
+    for repo in missing_beta_el8_repos:
+        assert f"Error: '{repo}' does not match a valid repository ID" in result.stdout
+    result = sat_maintain.execute('yum repolist')
+    for repo in sat_beta_repo:
+        assert repo in result.stdout
+
+
+@pytest.mark.capsule_only
+def test_positive_capsule_repositories_setup(sat_maintain):
+    """Verify that all required capsule repositories gets enabled.
+
+    :id: 88558fb0-2268-469f-86ae-c4d18ccef782
+
+    :parametrized: yes
+
+    :steps:
+        1. Run satellite-maintain advanced procedure run repositories-setup --version 6.y
+
+    :BZ: 1684730, 1869731
+
+    :expectedresults: Required Capsule repositories should get enabled
+    """
+    supported_versions = ['6.11', '6.12']
+    for ver in supported_versions:
+        result = sat_maintain.cli.Advanced.run_repositories_setup(options={'version': ver})
+        assert 'FAIL' not in result.stdout
+        assert result.status == 0
+        result = sat_maintain.execute('yum repolist')
+        for repo in cap_repos[ver]:
+            assert repo in result.stdout
+
+    # 6.13 till not GA
+    result = sat_maintain.cli.Advanced.run_repositories_setup(options={'version': '6.13'})
+    assert 'FAIL' in result.stdout
+    assert result.status == 1
+    for repo in cap_repos['6.13']:
+        assert repo in result.stdout
+
+    # Verify that all required beta repositories gets enabled
```

Comment:
Same as above for separate test.

Reply:
Same as above!

Reply:
No, there isn't a requirement for separate test, its good to have single test for GA and beta repo-setup